src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2662,22 +2659,12 @@ public void EndConnect(IAsyncResult asyncResult)
             EndPoint remoteEndPoint = null;
             ConnectOverlappedAsyncResult coar;
             MultipleAddressConnectAsyncResult macar;
-            ConnectAsyncResult car;
 
             coar = asyncResult as ConnectOverlappedAsyncResult;
             if (coar == null)
             {
                 macar = asyncResult as MultipleAddressConnectAsyncResult;
-                if (macar == null)
```

Comment:
No: a caller could pass in an invalid `IAsyncResult` that is neither a `ConnectOverlappedAsyncResult` nor a `MultipleAddressConnectAsyncResult`. In this case, `macar` would be `null` (and the check on [line 2692](https://github.com/dotnet/corefx/pull/3644/files#diff-a92dcc50fa7799a1b77b747a99c6222bL2692) would fail).